main.nf

```diff
@@ -314,11 +402,103 @@ def check_log(logs) {
         true
     }
 }
-// Filter removes all 'aligned' channels that fail the check
-aligned
-    .filter { logs, bams -> check_log(logs) }
-    .flatMap {  logs, bams -> bams }
+if(params.aligner == 'star'){
+    process star {
+        tag "$reads"
+        publishDir "${params.outdir}/STAR", mode: 'copy'
+
+        cpus { params.star_cpus ?: 10 }
+        memory { (params.star_memory ?: 80.GB) * task.attempt }
+        time { (params.star_time ?: 5.h) * task.attempt }
+        errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+
+        input:
+        file index from star_index
+        file annotation from gtf
+        file reads from trimmed_reads
+
+        output:
+        set file('*Log.final.out'), file ('*.bam') into star_aligned
+        file '*.out' into alignment_logs
+        file '*SJ.out.tab'
+
+        script:
+        """
+        #Getting STAR prefix
+        f=('$reads');f=\${f[0]};f=\${f%.gz};f=\${f%.fastq};f=\${f%.fq};f=\${f%_val_1};f=\${f%_trimmed};f=\${f%_1};f=\${f%_R1}
+        STAR --genomeDir $index \\
+            --sjdbGTFfile $annotation \\
+            --readFilesIn $reads  \\
+            --runThreadN ${task.cpus} \\
+            --twopassMode Basic \\
+            --outWigType bedGraph \\
+            --outSAMtype BAM SortedByCoordinate \\
+            --readFilesCommand zcat \\
+            --outFileNamePrefix \$f
+        """
+    }
+    // Filter removes all 'aligned' channels that fail the check
+    star_aligned
+        .filter { logs, bams -> check_log(logs) }
+        .flatMap {  logs, bams -> bams }
     .into { bam_count; bam_rseqc; bam_preseq; bam_markduplicates; bam_featurecounts; bam_stringtieFPKM }
+}
+
+
+/*
+ * STEP 3 - align with HISAT2
+ */
+if(params.aligner == 'hisat2'){
+    process hisat2 {
+        tag "$reads"
+        publishDir "${params.outdir}/HISAT2", mode: 'copy'
+
+        cpus { params.star_cpus ?: 10 }
+        memory { (params.star_memory ?: 80.GB) * task.attempt }
```

Comment:
I've actually removed `params.star_memory` in the next pull request that I'm about to open 😉 

But yes, we need to do proper profiling soon...